### PR TITLE
Add $dateToString opeartor

### PR DESCRIPTION
**Pull request by creek0810**
add basic datetostring operator

### CODE REVIEW

File: mongomock/aggregate.py [hunk 1]
```diff
@@ -358,6 +358,15 @@ def _handle_date_operator(self, operator, values):
...
         if operator == '$millisecond':
             return int(out_value.microsecond / 1000)
+        if operator == '$dateToString':
+            if '%L' in out_value['format']:
+                raise NotImplementedError(
+                    'Although %L is a valid date format for the '
+                    '$dateToString operator, it is currently not implemented '
+                    ' in Mongomock.'
+                )
+            return out_value['date'].strftime(out_value['format'])
```
Comment (pcorpet) [inline mongomock/aggregate.py:390]: What if `date` or `format` parameter is missing?

File: mongomock/aggregate.py [hunk 2]
```diff
@@ -358,6 +358,15 @@ def _handle_date_operator(self, operator, values):
             return out_value.second
         if operator == '$millisecond':
             return int(out_value.microsecond / 1000)
+        if operator == '$dateToString':
+            if '%L' in out_value['format']:
```
Comment (pcorpet) [inline mongomock/aggregate.py:372]: Please also add `NotImplementedError` when using `onNull` or `timezone` parameters.

**Review by pcorpet — CHANGES_REQUESTED** [review]
Please check #578, there are some comments that are still valid here.

File: mongomock/aggregate.py (hunk 2 above)
Comment (creek0810) [reply mongomock/aggregate.py:372]: done

File: mongomock/aggregate.py (hunk 1 above)
Comment (creek0810) [reply mongomock/aggregate.py:390]: add code to raise OperationFailure

**Comment by creek0810** [comment]
have added some code to handle OperationFailure and NotImplementedError condition

File: mongomock/aggregate.py [hunk 3]
```diff
@@ -358,6 +358,37 @@ def _handle_date_operator(self, operator, values):
             return out_value.second
         if operator == '$millisecond':
             return int(out_value.microsecond / 1000)
+        if operator == '$dateToString':
+            if not isinstance(values, dict):
+                raise OperationFailure(
+                    '$dateToString operator must correspond a dict'
+                    'that has "format" and "date" field.'
+                )
+            if 'format' not in values.keys() or 'date' not in values.keys():
```
Comment (pcorpet) [inline mongomock/aggregate.py]: Let's move that above:
```py
if not isinstance(values, dict) or not {'format', 'date'} <= set(values.keys()):
```

Comment (pcorpet) [reply mongomock/aggregate.py]: Actually `set(values)` instead of `set(values.keys())`

File: mongomock/aggregate.py [hunk 4]
```diff
@@ -358,6 +358,37 @@ def _handle_date_operator(self, operator, values):
...
+                    '$dateToString operator must correspond a dict'
+                    'that has "format" and "date" field.'
+                )
+            if '%L' in out_value['format']:
+                raise NotImplementedError(
+                    'Although %L is a valid date format for the '
+                    '$dateToString operator, it is currently not implemented '
+                    ' in Mongomock.'
+                )
+            if 'onNull' in values.keys():
```
Comment (pcorpet) [inline mongomock/aggregate.py]: No need to check `.keys()`, just `if 'onNull' in values:`

**Review by pcorpet — CHANGES_REQUESTED** [review]
Only cosmetic changes requested.

File: mongomock/aggregate.py (hunk 3 above)
Comment (creek0810) [reply mongomock/aggregate.py]: Done

File: mongomock/aggregate.py (hunk 4 above)
Comment (creek0810) [reply mongomock/aggregate.py]: Done

**Pull request merged**